### ISSUE TITLE
V12 beta: Simulation with 12.0.381 gives different output than simulation with version 12.0.372 without re-building the simulation

### ISSUE THREAD

**Issue by Christoph27** [Open-Systems-Pharmacology/MoBi#1688]
A simulation was built with Version 12.0.372 (and previous builds of V12 beta), cf.  project testPr6.zip attached below.

The output of the simulations (SimTest) looks like this - simulated with version 12.0.372 :

<img width="415" alt="372png" src="https://github.com/user-attachments/assets/b6d59116-1ef5-4ffd-8f5e-302774485b4d">

 If simulating the simulation with the beta version from today - MoBi 12.0.381 - (just opening the file and re-run the simulation without re-building it!), the output looks like this - both amounts are 0: 

<img width="413" alt="381" src="https://github.com/user-attachments/assets/b049facd-409f-4e70-99c2-9e05eba43722">

The output was more plausible to me before. Especially since "TargetCells" is immediately 0 with the version 381 despite an initial value > 0.  

[testPr6.zip](https://github.com/user-attachments/files/17918297/testPr6.zip)

**Comment by Yuri05** [comment]
@Christoph27 In one of the latest Mobi 12 builds we have set all values in the range `[0.. Absolute Tolerance]` to zero 
(to get rid of ugly plots in log scale if we have very small values over many orders of magnitude)

Looking at you plot: simulated values are extremely small (1e-16..1e-14) ant for sure below your absolute tolerance.
So either you decrease your absolute tolerance (e.g. to 1e-17) - or we need to roll back this change...
![{F5AC8674-D533-4C51-B91E-368E18C7B8D7}](https://github.com/user-attachments/assets/c8af0519-310e-4a62-9786-4c5de6472104)

**Comment by Christoph27** [comment]
The numbers are supposed to be that small. I use "Calculate Scale Divisor" to work with them. Having scaled these state variables, shouldn’t it be OK to work with them (without decreasing absolute tolerance to the smallest numbers for all state variables)?  At least before all worked fine with these small numbers without any "ugly plots" after applying "Calculate Scale Divisor".

**Comment by Yuri05** [comment]
> Having scaled these state variables, shouldn’t it be OK to work with them (without decreasing absolute tolerance to the smallest numbers for all state variables)?

Yes, this should be OK. **However, in the attached example, all scale divisors in the simulation are set to 1.** 

I have set the scale divisors for the 2 variables from the screenshot and rerun the simulation.
Then the result looked good again
![{4EB9ACD4-88B7-4500-9DAD-509ED8A78D08}](https://github.com/user-attachments/assets/0417c7f9-fb77-4b70-9e77-6479a66f223a)

![grafik](https://github.com/user-attachments/assets/1e94f4d3-eeb2-4ebc-8b5b-8e6062e2dff7)

**Comment by Christoph27** [comment]
OK, it seems I freshly rebuild the simulation from the building block before uploading to GitHub, so that the scale divisors were set to 0 again.

What I have done is to use “Calculate Scale Divisors” (not setting the scale divisors manually) and the problem can be reproduced. It seems that “Calculate Scale divisors” is not working properly anymore: 

With build 372, the scale divisors are small numbers also for “TargetCells” and “deadTargetCells_byADCP” as expected:

![SD_372](https://github.com/user-attachments/assets/e0f2551b-77c8-43dc-9468-186408200f33)


With the newer build, the scale divisors for “TargetCells” and ““deadTargetCells_byADCP” are 1 even after “Calculate Scale Divisors” for whatever reason: 

![SD_new](https://github.com/user-attachments/assets/b8dbace8-42a3-4b53-a461-0567c0b9bd45)

Since I used “Calculate Scale Divisors” I got – and still get – the “all-0” profiles.

**Comment by Yuri05** [comment]
Well, it's kind of chicken egg problem.
As the simulated values (calculated with the initial scale divisors = 1) are const. zero - calculating the scale divisors based on these simulation results will return 1 for scale divisors again.

Need to think how to deal with this...

**Comment by msevestre** [comment]
I was about to say the same  
But turkey and egg problem since it’s thanksgiving in the USA today ;)

**Comment by Christoph27** [comment]
Setting to 0 should only be done for state variables which are going to be plotted and before a plotting step. 
Why should “Calculate Scale Divisors” care of this? Shouldn't “Calculate Scale Divisors” be just based on values before setting to 0?

**Comment by Yuri05** [comment]
> Setting to 0 should only be done for state variables which are going to be plotted and before a plotting step. Why should “Calculate Scale Divisors” care of this? Shouldn't “Calculate Scale Divisors” be just based on values before setting to 0?

It's more complicated than this. To know the threshold (below which the values can be considered 0) you need to take into account the effectively used absolute tolerance (which might differ from the passed one 😉), used scale divisor and other eventual factors (e.g. if you plot a _concentration_ you should divide the threshold of the state variable (amount) by the corresponding volume as well). 

Well, we do calculate this threshold for all state variables and observers of the system, however it is not propagated to the plots.
Could be done, but not V12.

For now, the only possibility I see is to roll back the change which causes the issue.
@msevestre Or what do you think?
